### DISCUSSION TITLE
Any idea of why the bootstrap carousel isn't moving?

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#41031] [Help] [answered]
This is the code I have... I am using Bootstrap 5.1.3

```
<div className="row tu-world">
        <div
          id="carouselExampleSlidesOnly"
          className="carousel slide"
          data-bs-ride="carousel"
        >
          <div className="carousel-inner">
            <div className="carousel-item active">
              <Image
                className="d-block w-100"
                src="/carousel-main/tu-world.png"
                alt="tu world"
                width={1710}
                height={315}
              />
            </div>
            <div className="carousel-item">
              <Image
                className="d-block w-100"
                src="/carousel-main/carousel-pumps.png"
                alt="hydraulic pumps"
                width={1710}
                height={315}
              />
            </div>
            <div className="carousel-item ">
              <Image
                className="d-block w-100"
                src="/carousel-main/carousel-everything.png"
                alt="everything industrial"
                width={1710}
                height={315}
              />
            </div>
          </div>
        </div>
```

The Bootstrap documentation is this:
https://getbootstrap.com/docs/5.1/components/carousel/

I am also getting a few errors at the console... I assume they are related to the adsense code that is always bugging with errors..

```

service worker installed
adsbygoogle.js?client=ca-pub-8682017023658879:252 AdSense head tag doesn't support data-nscript attribute.
(anonymous) @ adsbygoogle.js?client=ca-pub-8682017023658879:252
ads:1          Failed to load resource: the server responded with a status of 403 ()
ads:1          Failed to load resource: the server responded with a status of 403 ()
2sw.js:1 Uncaught (in promise) DOMException: Quota exceeded.

```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

The issue is that you need to use JS to control the class names applied to each item of the carousel. For example the `active` class name has to be passed around between each carousel-item, as you also need to apply briefly transition class names to the entering item. 

In short, there's a reason why react-bootstrap exists, and you actually have it on your project.

Here's the documentation for it: https://react-bootstrap.github.io/components/carousel/

```jsx
// at the top of your file add this import where you see most fit
import Carousel  from "react-bootstrap/Carousel";


// later on the JSX
<div className="row tu-world">
  <Carousel>
    <Carousel.Item>
      <Image
        className="d-block w-100"
        src="/carousel-main/tu-world.png"
        alt="tu world"
        width={1710}
        height={315}
      />
    </Carousel.Item>
    <Carousel.Item>
      <Image
        className="d-block w-100"
        src="/carousel-main/carousel-pumps.png"
        alt="hydraulic pumps"
        width={1710}
        height={315}
      />
    </Carousel.Item>
    <Carousel.Item>
      <Image
        className="d-block w-100"
        src="/carousel-main/carousel-everything.png"
        alt="everything industrial"
        width={1710}
        height={315}
      />
    </Carousel.Item>
  </Carousel>
</div>
```

Note that it is entirely possible to this manually, with plain React, and with any pattern we'd like to, but why re-invent the wheel. There's also the `bootstrap` script we could bring into scope to work with their API, but again, you've got `react-bootstrap` installed, so use it.

By the way the https://getbootstrap.com/docs/5.1/components/carousel/#how-it-works docs are a bit subtle about this:

> How it works
> The carousel is a slideshow for cycling through a series of content, built with CSS 3D transforms **_and a bit of JavaScript._** It works with a series of images, text, or custom markup. It also includes support for previous/next controls and indicators.

**Reply by Elindo586** [reply]
Oh I see... I should be working with react-bootstrap instead...    I had regular bootstrap from the previous website conversion before React...

I'll give it a try..

**Reply by Aastha-Kalra** [reply]
Hey there, i am using react-bootstrap but my prev and next buttons are not working also not showing any error. 
what should i do?

**Reply by Aastha-Kalra** [reply]
1. > //my code

import  Carousel  from "react-bootstrap/Carousel";
<div className="container-fluid bg-dark mb-5 p-4">
       <span style={{fontSize: 5+'em', color:'grey', position:'absolute' , top:35+'em', left:7+'em'}}><VscActivateBreakpoints/></span>
      <h1 className="text-center pt-5 ">
            Activities</h1>
<Carousel  style={{overflow:'hidden',position:'relative',padding:23+'px'}} >
      <Carousel.Item  style={{paddingInlineStart:23+'px' , display:'flex'}}>
      {image.map((image,i)=>{
            return <div className="card1 m-1 float-start" style={{padding:7+'px'}}>
                  <div key={i}>
                        <img src={image.Src} alt="" />
                  </div>
            </div>
      })
}
      </Carousel.Item>
</Carousel>
    
</div>
